include/swift/AST/Types.h

```diff
@@ -39,6 +39,25 @@
 #include "llvm/Support/ErrorHandling.h"
 #include "llvm/Support/TrailingObjects.h"
 
+namespace clang {
+class Type;
+class FunctionType;
+} // namespace clang
+
+#ifndef NDEBUG
+#include "clang/AST/Type.h"
+namespace {
+void checkIsFunctionType(const clang::Type *type) {
```

Comment:
We usually just declare helper functions like this `static` (`static inline` since this is a header) instead of putting them in an anonymous namespace.  But actually I don't think it's a good idea to define this in the header, *especially* since it conditionally includes a major header, which exposes us to all sorts of risk where code will compile in NDEBUG builds but not otherwise; just define it out-of-line.  Also consider making it a static function on ExtInfo, which is the only place that uses it.